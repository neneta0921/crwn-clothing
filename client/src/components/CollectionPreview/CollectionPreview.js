import { withRouter } from 'react-router-dom';

import CollectionItem from '../CollectionItem/CollectionItem';
import {
  CollectionPreviewContainer,
  PreviewContainer,
  TitleContainer,
} from './CollectionPreviewStyles';

export const CollectionPreview = ({ title, items, history, match, routeName }) => (
  <CollectionPreviewContainer>
    <TitleContainer
      className="title-container"
      onClick={() => history.push(`${match.path}/${routeName}`)}>
      {title.toUpperCase()}
    </TitleContainer>
    <PreviewContainer>
      {items
        .filter((item, idx) => idx < 4)
        .map((item) => (
          <CollectionItem key={item.id} item={item} />
        ))}
    </PreviewContainer>
  </CollectionPreviewContainer>
);

export default withRouter(CollectionPreview);
