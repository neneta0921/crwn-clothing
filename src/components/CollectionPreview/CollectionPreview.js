import { CollectionItem } from '../CollectionItem/CollectionItem';
import './CollectionPreview.scss';

export const CollectionPreview = ({ title, items }) => (
  <div className='collection-preview'>
    <h2 className='title'>{title.toUpperCase()}</h2>
    <div className='preview'>
      {items
        .filter((item, idx) => idx < 4)
        .map(({id, ...otherItemProps}) => (
          <CollectionItem key={id} {...otherItemProps} />
      ))}
    </div>
  </div>
)