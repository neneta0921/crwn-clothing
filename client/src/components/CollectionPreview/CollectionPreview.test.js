import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { CollectionPreview } from './CollectionPreview';

describe('CollectionPreview component', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const mockRouteName = 'hats';

  beforeEach(() => {
    mockMatch = { path: '/shop' };
    mockHistory = { push: jest.fn() };
    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      routeName: mockRouteName,
      title: 'hats',
      items: [],
    };

    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  it('should render CollectionPreview component', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call history.push with the right string when TitleContainer clicked', () => {
    wrapper.find('.title-container').simulate('click');
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.path}/${mockRouteName}`);
  });
});
