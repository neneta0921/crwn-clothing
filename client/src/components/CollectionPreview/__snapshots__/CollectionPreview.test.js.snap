// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component should render CollectionPreview component 1`] = `
<div
  className="sc-iBPRYJ gyKNFV"
>
  <h2
    className="sc-fubCfw bLzYkF title-container"
    onClick={[Function]}
  >
    HATS
  </h2>
  <div
    className="sc-pFZIQ grNlYC"
  />
</div>
`;
